pkg/apis/internal.acorn.io/v1/memory.go

```diff
@@ -0,0 +1,47 @@
+package v1
+
+import (
+	"strconv"
+
+	"cuelang.org/go/cue/literal"
+)
+
+func ParseMemoryMap(memory map[string]string) (Memory, error) {
+	resources := Memory{
+		All:       0,
+		Workloads: map[string]int64{},
+	}
+	allSet := false
```

Comment:
Can you explain whats going on with this allSet logic? I'm confused

Reply:
This is how we allow a user to do `acorn run -m 512Mi` and `acorn run -m nginx:512Mi` with the same flag.

Reply:
This function got heavily overhauled. Mind giving it another look and seeing if its still confusing? The approach is all new.